### PR TITLE
Smart Wallets: User Based Authentication (Closes WAL-2430)

### PR DESCRIPTION
**Pull request by devlyn37**
## Description

This PR adds user based authentication for calls to internal SDK endpoints. Depends on https://github.com/Paella-Labs/crossbit-main/pull/13736

## Test plan

Locally, using JWTs created within the SCW Demo site, I:
- Created a new wallet and minted.
- Went through the minting flow in a new session with the same user.

More testing will be done during Passkey release QA.

### CODE REVIEW

File: packages/client/wallets/aa/src/types/API.ts
Comment (devlyn37) [inline packages/client/wallets/aa/src/types/API.ts:6]: We now grab the user from the JWT passed in.

File: packages/client/wallets/aa/src/api/CrossmintWalletService.ts
Comment (devlyn37) [inline packages/client/wallets/aa/src/api/CrossmintWalletService.ts:10]: Check https://github.com/Paella-Labs/crossbit-main/pull/13736/files for more context on these changes

File: packages/client/wallets/aa/src/api/BaseCrossmintService.ts
Comment (maxisch) [inline packages/client/wallets/aa/src/api/BaseCrossmintService.ts:50]: nit: It's my understanding that the convention is sending `Bearer` with capital `B`

File: packages/client/wallets/aa/src/api/CrossmintWalletService.ts
Comment (maxisch) [inline packages/client/wallets/aa/src/api/CrossmintWalletService.ts:66]: just double checking here. The passkeys server won't be auth protected for the first release, right? Because of the hardcoded headers issue?

**Review by maxisch — APPROVED** [review]

File: packages/client/wallets/aa/src/api/CrossmintWalletService.ts
Comment (devlyn37) [reply packages/client/wallets/aa/src/api/CrossmintWalletService.ts:66]: Yeah! Looks like the security review is going to block the release though so we may have time before release though

File: packages/client/wallets/aa/src/api/BaseCrossmintService.ts
Comment (devlyn37) [reply packages/client/wallets/aa/src/api/BaseCrossmintService.ts]: fixed, ty for catching

**Pull request merged**
